### ISSUE TITLE
Add support for footnotes lint

### ISSUE THREAD

**Issue by gagahpangeran** [ppy/osu-wiki#6232]
Because https://github.com/ppy/osu-web/pull/8125 is already merged (and waiting to be deployed), I was testing [my footnote branch](https://github.com/gagahpangeran/osu-wiki/tree/footnote) ([diff](https://github.com/ppy/osu-wiki/compare/master...gagahpangeran:footnote?expand=1)) with current linter.

Looks like current linter does not support footnotes syntax yet.

<details>
<summary>error</summary>

```
$ npx remark -qf --no-stdout --silently-ignore --report=vfile-reporter-position --color wiki/Article_styling_criteria/Formatting/en.md
wiki/Article_styling_criteria/Formatting/en.md
  ✖ wiki/Article_styling_criteria/Formatting/en.md:9:48 - error remark-lint:no-undefined-references Found reference to undefined definition

  9 Listed below are the properly-supported locales[^locales] for the wiki:
                                                   ~~~~~~~~~~

  ✖ wiki/Article_styling_criteria/Formatting/en.md:56:7 - error remark-lint:no-undefined-references Found reference to undefined definition

  56 - Tags[^tags]
           ~~~~~~~

  ✖ wiki/Article_styling_criteria/Formatting/en.md:61:13 - error remark-lint:no-undefined-references Found reference to undefined definition

  61 Front matter[^cite] must be placed at the very top of the file. It is written in [YAML](https://en.wikipedia.org/wiki/YAML#Example "YAML Wikipedia article") and describes additional information about the article. This must be surrounded by three hyphens (`---`) on the lines above and below it, and an empty line must follow it before the title heading.
                 ~~~~~~~

  ✖ wiki/Article_styling_criteria/Formatting/en.md:67:88 - error remark-lint:no-undefined-references Found reference to undefined definition

  67 The `needs_cleanup` tag may be added to articles that need rewriting or formatting help[^help]. It is also acceptable to open an issue on GitHub for this purpose. This tag must be written as shown below:
                                                                                            ~~~~~~~

  ✖ wiki/Article_styling_criteria/Formatting/en.md:79:91 - error remark-lint:no-undefined-references Found reference to undefined definition

  79 Translated articles that are outdated must use the `outdated` tag when the English variant[^variant] is updated. English articles may also become outdated when the content they contain is misleading or no longer relevant. This tag must be written as shown below:
                                                                                               ~~~~~~~~~~

  ✖ wiki/Article_styling_criteria/Formatting/en.md:89:38 - error remark-lint:no-undefined-references Found reference to undefined definition

  89 Tags help the website's search engine[^search] query articles better. Tags should be written in the same language as the article and include the original list of tags. Tags should use lowercase letters where applicable.
                                          ~~~~~~~~~

  ✖ wiki/Article_styling_criteria/Formatting/en.md:776:1 - error remark-lint:no-undefined-references Found reference to undefined definition

  776 [^locales]: This list is based on [ISO 639-1](https://en.wikipedia.org/wiki/ISO_639-1)
      ~~~~~~~~~~

  ✖ wiki/Article_styling_criteria/Formatting/en.md:777:1 - error remark-lint:no-undefined-references Found reference to undefined definition

  777 [^tags]: <https://www.urbandictionary.com/define.php?term=Tags> *Be careful to open this link*
      ~~~~~~~

  ✖ wiki/Article_styling_criteria/Formatting/en.md:778:1 - error remark-lint:no-undefined-references Found reference to undefined definition

  778 [^cite]: ***Need citation!!!***
      ~~~~~~~

  ✖ wiki/Article_styling_criteria/Formatting/en.md:779:1 - error remark-lint:no-undefined-references Found reference to undefined definition

  779 [^help]: Help!!! Help!!!
      ~~~~~~~

  ✖ wiki/Article_styling_criteria/Formatting/en.md:780:1 - error remark-lint:no-undefined-references Found reference to undefined definition

  780 [^variant]: *What is variant?*
      ~~~~~~~~~~

  ✖ wiki/Article_styling_criteria/Formatting/en.md:781:1 - error remark-lint:no-undefined-references Found reference to undefined definition

  781 [^search]: For example google
      ~~~~~~~~~


12 errors and 0 warnings
```

</details>

Quick solution that I tried is upgrading `remark-gfm` to version `3.0.0` because [it already has support for footnotes syntax](https://github.com/remarkjs/remark-gfm/releases/tag/3.0.0). But I also need to upgrade `remark-cli` to version `10.0.0`, otherwise it gives this weird error.

<details>
<summary>weird error</summary>

```
$ npx remark -qf --no-stdout --silently-ignore --report=vfile-reporter-position --color wiki/Article_styling_criteria/Formatting/en.md
wiki/Article_styling_criteria/Formatting/en.md
  ✖ wiki/Article_styling_criteria/Formatting/en.md - error Cannot read property '4' of undefined

1 error and 0 warnings
```

</details>

**tl;dr: we need to upgrade `remark-gfm` and `remark-cli` to latest version to support footnotes syntax lint.**

*ps: maybe while we at it, we could also upgrade another dependencies too. (but need further testing to make sure the CI not broken)*

---

Just drop this here for now. Maybe we should do this when there's article that use footnotes and the CI start to failing.

**Comment by TicClick** [comment]
perhaps I have done something wrong, but I can't replicate your errors on an article with footnotes edited in (`nvm v10.24.1, npm v6.14.12, remark v13.0.0, remark-cli v9.0.0`).

I also want to confess I don't have the slightest idea about node.js' ecosystem, but I think our packages are long overdue for updating, INCLUDING `node` itself in our GitHub actions (for the reference, it's 10.24.1, which [isn't even covered by the LTS](https://nodejs.org/en/about/releases/)).

so, can we just..? @cl8n what do you think?
